src/main/scala/io/iohk/ethereum/network/AuthHandshaker.scala

```diff
@@ -15,12 +15,12 @@ import org.spongycastle.math.ec.ECPoint
 import scorex.core.network
 import scorex.core.network._
 
-class Secrets(
-    val aes: Array[Byte],
-    val mac: Array[Byte],
-    val token: Array[Byte],
-    val egressMac: KeccakDigest,
-    val ingressMac: KeccakDigest)
+case class Secrets(
```

Comment:
I see a small problem with this. A `case class` implements `equals` and `hashCode` et. al. with the intent of storing and comparison of immutable data. However because this one consists of `Arrays` it will not work:

```
scala> case class A(arr: Array[Int])
defined class A

scala> A(Array(1)) == A(Array(1))
res0: Boolean = false
```

Maybe it would be better to leave it as a regular class so as not give a false impression?
Alternatively maybe those secrets could be stored within immutable structures like `ByteString`?